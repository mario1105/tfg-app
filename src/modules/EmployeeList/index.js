import React from 'react'
import {
    Typography, makeStyles, Button
} from '@material-ui/core';
import ReturnButton from "../../components/ReturnButton";

const useStyles = makeStyles(theme => ({
    employeesBox: {
        backgroundColor: theme.app.background['300'],
        padding: theme.spacing(7),
        borderRadius: 10,
        margin: '10em 40em 10em 40em',
    }
}))

const EmployeeList = ({ location }) => {
    const user = location.state

    const classes = useStyles()

    return (
        <div className={classes.employeesBox}>
            <ReturnButton previousRoute="dashboard" user={user} />
        </div>
    )
}

export default EmployeeList

