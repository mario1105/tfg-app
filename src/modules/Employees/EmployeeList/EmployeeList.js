import React from 'react'
import TableDataRowWrapper from "../../../components/TableDataRowWrapper"
import TableRowHeaders from "../../../components/TableRowHeaders"
import EmployeeListItemContainer from "./EmployeeListItem"

const EmployeeList = ({ role, employees, handleRemoveEmployee }) => {
    const headers = ['First Name', 'Last Name', 'Email', 'Date of Birth', 'Mobile phone']
    return (
        <>
            <TableRowHeaders
                headerList={ role === 'user' ? headers : [...headers, 'Salary', 'Actions'] }
            />
            {employees.map(
                ({ id, firstName, lastName, email, dateOfBirth, mobilePhone, salary }) => (
                    <TableDataRowWrapper key={id}>
                        <EmployeeListItemContainer
                            id={id}
                            role={role}
                            firstName={firstName}
                            lastName={lastName}
                            email={email}
                            dateOfBirth={dateOfBirth}
                            mobilePhone={mobilePhone}
                            salary={salary}
                            handleRemoveEmployee={handleRemoveEmployee}
                        />
                    </TableDataRowWrapper>
                )
            )}
        </>
    )
}

export default EmployeeList

