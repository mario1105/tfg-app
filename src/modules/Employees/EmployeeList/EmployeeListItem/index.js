import React, {useEffect, useState} from 'react'
import EmployeeListItem from "./EmployeeListItem"
import employeesService from "../../../../services/employeesService"

const EmployeeListItemContainer = ({ id, role, firstName, lastName, email, dateOfBirth, mobilePhone, salary, handleRemoveEmployee }) => {
    const [parameters, setParameters] = useState({firstName, lastName, email, dateOfBirth, mobilePhone, salary})
    const [editMode, setEditMode] = useState(false)

    const handleOnChange = e => {
        setParameters({
            ...parameters,
            [e.target.name]: e.target.value
        })
    }

    const handleEditMode = () => setEditMode(!editMode)

    const handleOnSubmit = async () => {
       await employeesService.editEmployee({parameters, id})
        handleEditMode()
    }

    const handleOnRemove = async () => {
        await employeesService.removeEmployee({id})
        handleRemoveEmployee(id)
    }

    return (
        <EmployeeListItem
            id={id}
            role={role}
            parameters={parameters}
            editMode={editMode}
            handleOnChange={handleOnChange}
            handleEditMode={handleEditMode}
            handleOnSubmit={handleOnSubmit}
            handleOnRemove={handleOnRemove}
        />
    )
}

export default EmployeeListItemContainer

