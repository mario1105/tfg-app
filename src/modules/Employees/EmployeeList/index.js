import React, { useEffect, useState } from 'react'
import * as R from 'ramda'
import EmployeeList from "./EmployeeList"
import employeesService from "../../../services/employeesService"

const EmployeeListContainer = ({ role }) => {
    const [employees, setEmployees] = useState([])

    const handleRemoveEmployee = (id) => setEmployees(R.filter((employee) => employee.id !== id, employees))

    useEffect(() => {
        const fetchData = async () => {
            const response = await employeesService.getEmployees()
            setEmployees(response)
        }
        fetchData()
    }, [])

    return (
        <EmployeeList role={role} employees={employees} handleRemoveEmployee={handleRemoveEmployee} />
    )
}

export default EmployeeListContainer

