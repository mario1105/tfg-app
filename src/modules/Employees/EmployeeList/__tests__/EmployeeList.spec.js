import React from 'react'
import { shallow } from 'enzyme'
import EmployeeList from "../EmployeeList"

const props = {
    role: 'admin',
    employees: [
        {
            id: 1,
            firstName: 'Mario',
            lastName: 'Jimenez',
            email: 'mario@test.com',
            dateOfBirth: '11/05/1995',
            mobilePhone: '666555444',
            salary: '355000'
        },
        {
            id: 2,
            firstName: 'Daniel',
            lastName: 'Gomez',
            email: 'daniel@test.com',
            dateOfBirth: '19/11/1992',
            mobilePhone: '612345632',
            salary: '42000'
        },
        {
            id: 3,
            firstName: 'Carlos',
            lastName: 'Lopez',
            email: 'carlos@test.com',
            dateOfBirth: '25/02/1996',
            mobilePhone: '692746285',
            salary: '30000'
        }
    ],
    handleRemoveEmployee: () => null
}
const renderComponent = () => {
    return (shallow(<EmployeeList {...props} />))
}

let wrapper
describe('EmployeeList Unit Tests', () => {
    it('renders as expected', () => {
        wrapper = renderComponent()
        expect(wrapper).toMatchSnapshot()
    })
})
