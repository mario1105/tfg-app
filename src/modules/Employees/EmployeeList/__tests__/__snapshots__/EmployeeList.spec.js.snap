// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeList Unit Tests renders as expected 1`] = `
<Fragment>
  <TableRowHeaders
    headerList={
      Array [
        "First Name",
        "Last Name",
        "Email",
        "Date of Birth",
        "Mobile phone",
        "Salary",
        "Actions",
      ]
    }
  />
  <TableDataRowWrapper
    key="1"
  >
    <EmployeeListItemContainer
      dateOfBirth="11/05/1995"
      email="mario@test.com"
      firstName="Mario"
      handleRemoveEmployee={[Function]}
      id={1}
      lastName="Jimenez"
      mobilePhone="666555444"
      role="admin"
      salary="355000"
    />
  </TableDataRowWrapper>
  <TableDataRowWrapper
    key="2"
  >
    <EmployeeListItemContainer
      dateOfBirth="19/11/1992"
      email="daniel@test.com"
      firstName="Daniel"
      handleRemoveEmployee={[Function]}
      id={2}
      lastName="Gomez"
      mobilePhone="612345632"
      role="admin"
      salary="42000"
    />
  </TableDataRowWrapper>
  <TableDataRowWrapper
    key="3"
  >
    <EmployeeListItemContainer
      dateOfBirth="25/02/1996"
      email="carlos@test.com"
      firstName="Carlos"
      handleRemoveEmployee={[Function]}
      id={3}
      lastName="Lopez"
      mobilePhone="692746285"
      role="admin"
      salary="30000"
    />
  </TableDataRowWrapper>
</Fragment>
`;
